sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -88,6 +87,22 @@ class RapidsGdsStore(
       }
     }
 
+    override def copyToMemoryBuffer(
+        srcOffset: Long, dst: MemoryBuffer, dstOffset: Long, length: Long): Unit = {
+      val path = if (id.canShareDiskPaths) {
+        sharedBufferFiles.get(id)
+      } else {
+        id.getDiskPath(diskBlockManager)
+      }
+      dst match {
+        case _: DeviceMemoryBuffer =>
+          val dm = dst.asInstanceOf[DeviceMemoryBuffer].slice(dstOffset, length)
+          CuFile.readFileToDeviceBuffer(dm, path, fileOffset + srcOffset)
```

Comment:
Curious, what stream is used here?  It looks like it's default, but it's unclear whether that's per-thread default or system default.

Reply:
nice catch, we should match the stream we are synchronizing on (the `BufferSendState` takes a stream)

Reply:
@rongou this PR is still held by this right?

Reply:
I asked the GDS team they said right now this is just synchronous. I added a TODO to switch to the async api when it becomes available.

Reply:
This seems fine to me for now.